### PR TITLE
Add List component for detail properties

### PR DESCRIPTION
**Pull request by jtomasek**
This adds common component for lists in detail page properties.
- Common styling
- Semantic use of lists instead of spans and brs

Screenshot 2020-01-08 at 09 33 27

### CODE REVIEW

File: frontend/packages/console-app/src/components/nodes/NodeIPList.tsx [hunk 1]
```diff
@@ -12,15 +12,13 @@ const NodeIPList: React.FC<NodeIPListProps> = ({ ips, expand = false }) => (
     {_.sortBy(ips, ['type']).map(
       ({ type, address }) =>
         address && (
-          <div key={`{${type}/${address}`} className="co-node-ip">
+          <DetailPropertyList key={`{${type}/${address}`}>
```
Comment (suomiy) [inline frontend/packages/console-app/src/components/nodes/NodeIPList.tsx]: shouldn't this be wrapped arround all ips? and not for each item

File: frontend/packages/console-shared/src/components/lists/DetailPropertyList.scss [hunk 2]
```diff
@@ -0,0 +1,12 @@
+ul.pf-c-list:not(.pf-m-inline).co-detail-property-list {
+  list-style-type: none;
+  list-style: none;
```
Comment (suomiy) [inline frontend/packages/console-shared/src/components/lists/DetailPropertyList.scss]: don't these two result in the similar thing? Maybe we can use just `list-style`

File: frontend/packages/console-shared/src/components/lists/DetailPropertyList.scss [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+  padding-left: 0;
+  li + li {
```
Comment (suomiy) [inline frontend/packages/console-shared/src/components/lists/DetailPropertyList.scss]: why do we need this?

File: frontend/packages/console-shared/src/components/lists/DetailPropertyListItem.tsx [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import * as React from 'react';
+import { ListItem } from '@patternfly/react-core';
+
+type Props = {
+  title?: string;
```
Comment (suomiy) [inline frontend/packages/console-shared/src/components/lists/DetailPropertyListItem.tsx]: \+ children?

File: frontend/packages/metal3-plugin/src/components/baremetal-hosts/BareMetalHostDetails.tsx [hunk 5]
```diff
@@ -96,11 +96,15 @@ const BareMetalHostDetails: React.FC<BareMetalHostDetailsProps> = ({
             )}
             <dt>Host Addresses</dt>
             <dd>
-              <ul className="metal3-bare-metal-host-details__simple-list">
-                <li>Management: {getHostBMCAddress(host)}</li>
-                <li>NICs: {ips}</li>
-                <li>Boot Interface MAC: {getHostBootMACAddress(host)}</li>
-              </ul>
+              <DetailPropertyList>
```
Comment (suomiy) [inline frontend/packages/metal3-plugin/src/components/baremetal-hosts/BareMetalHostDetails.tsx]: +1 easier to use

File: frontend/packages/console-shared/src/components/lists/DetailPropertyList.scss (hunk 3 above)
Comment (jtomasek) [reply frontend/packages/console-shared/src/components/lists/DetailPropertyList.scss]: This is to eliminate margin between `li`s defined in `.pf-c-list li+li ` rule.

File: frontend/packages/console-shared/src/components/lists/DetailPropertyList.scss (hunk 2 above)
Comment (jtomasek) [reply frontend/packages/console-shared/src/components/lists/DetailPropertyList.scss]: Right, removing.

File: frontend/packages/console-app/src/components/nodes/NodeIPList.tsx (hunk 1 above)
Comment (jtomasek) [reply frontend/packages/console-app/src/components/nodes/NodeIPList.tsx]: Oops, yes, I think the `div` was here by mistake as well as it should not render when the address is not supposed to render.

File: frontend/packages/console-shared/src/components/lists/DetailPropertyListItem.tsx (hunk 4 above)
Comment (jtomasek) [reply frontend/packages/console-shared/src/components/lists/DetailPropertyListItem.tsx]: `children` is already included in React.FC via React.PropsWithChildren type (https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react/index.d.ts#L518)

File: frontend/packages/console-shared/src/components/lists/DetailPropertyList.tsx [hunk 6]
```diff
@@ -0,0 +1,15 @@
+import * as React from 'react';
+import { List } from '@patternfly/react-core';
+
+import './DetailPropertyList.scss';
+
+const DetailPropertyList: React.FC<React.ComponentProps<typeof List>> = ({
```
Comment (rawagner) [inline frontend/packages/console-shared/src/components/lists/DetailPropertyList.tsx]: you can use `ListProps` directly
```suggestion
const DetailPropertyList: React.FC<ListProps> = ({
```

File: frontend/packages/console-shared/src/components/lists/DetailPropertyListItem.tsx [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+};
+
+const DetailPropertyListItem: React.FC<Props> = ({ title, children }) => (
+  <ListItem>
+    {title && <span className="co-detail-property-list__item-title">{title}: </span>}
```
Comment (rawagner) [inline frontend/packages/console-shared/src/components/lists/DetailPropertyListItem.tsx]: missing `import './DetailPropertyList.scss';` import. There would be an issue with missing styles if we use `DetailPropertyListItem` without `DetailPropertyList`

File: frontend/packages/console-shared/src/components/lists/DetailPropertyListItem.tsx [hunk 8]
```diff
@@ -0,0 +1,15 @@
+import * as React from 'react';
+import { ListItem } from '@patternfly/react-core';
+
+type Props = {
```
Comment (rawagner) [inline frontend/packages/console-shared/src/components/lists/DetailPropertyListItem.tsx]: can you name this better - such as `DetailPropertyListItemProps` and move to bottom as we usually do for type declarations ?

File: frontend/packages/console-shared/src/components/lists/DetailPropertyList.tsx (hunk 6 above)
Comment (jtomasek) [reply frontend/packages/console-shared/src/components/lists/DetailPropertyList.tsx]: Done

File: frontend/packages/console-shared/src/components/lists/DetailPropertyListItem.tsx (hunk 7 above)
Comment (jtomasek) [reply frontend/packages/console-shared/src/components/lists/DetailPropertyListItem.tsx]: Done

File: frontend/packages/console-shared/src/components/lists/DetailPropertyListItem.tsx (hunk 8 above)
Comment (jtomasek) [reply frontend/packages/console-shared/src/components/lists/DetailPropertyListItem.tsx]: I've renamed the props but I'd prefer to keep them on top. Both conventions are used in console and I find putting props to the top more convenient as it's not necessary to scroll over the component body to see the props.

**Pull request merged**